src/network/tarcap/packets_handler/handlers/dag_packets_handler.cpp

```diff
@@ -0,0 +1,230 @@
+  } else if (packet_data.type_ == NewBlockHashPacket) {
+    processNewBlockHashPacket(packet_data, packet_rlp);
+  } else if (packet_data.type_ == GetNewBlockPacket) {
+    processGetNewBlockPacket(packet_data, packet_rlp);
+  } else {
+    assert(false);
+  }
+}
+
+inline void DagPacketsHandler::processNewBlockPacket(const PacketData &packet_data, const dev::RLP &packet_rlp) {
+  // Ignore new block packets when syncing
+  if (syncing_state_->is_syncing()) return;
+
+  DagBlock block(packet_rlp[0].data().toBytes());
+  blk_hash_t const hash = block.getHash();
+
+  if (dag_blk_mgr_) {
+    if (dag_blk_mgr_->isBlockKnown(hash)) {
+      LOG(log_dg_) << "Received NewBlock " << hash.toString() << "that is already known";
+      return;
+    }
+    if (auto status = syncing_state_->checkDagBlockValidation(block); !status.first) {
+      LOG(log_wr_) << "Received NewBlock " << hash.toString() << " missing pivot or/and tips";
+      status.second.push_back(hash);
+      syncing_state_->requestBlocks(packet_data.from_node_id_, status.second,
+                                    GetBlocksPacketRequestType::MissingHashes);
+      return;
+    }
+  }
+
+  const auto transactions_count = packet_rlp.itemCount() - 1;
+  LOG(log_dg_) << "Received NewBlockPacket " << transactions_count;
+
+  std::vector<Transaction> new_transactions;
+  for (size_t i_transaction = 1; i_transaction < transactions_count + 1; i_transaction++) {
+    Transaction transaction(packet_rlp[i_transaction].data().toBytes());
+    new_transactions.push_back(transaction);
+    peer_->markTransactionAsKnown(transaction.getHash());
+  }
+
+  peer_->markBlockAsKnown(hash);
+  if (block.getLevel() > peer_->dag_level_) peer_->dag_level_ = block.getLevel();
+  onNewBlockReceived(block, new_transactions);
+}
+
+inline void DagPacketsHandler::processNewBlockHashPacket(const PacketData &packet_data, const dev::RLP &packet_rlp) {
+  blk_hash_t const hash(packet_rlp[0]);
+  LOG(log_dg_) << "Received NewBlockHashPacket " << hash.toString();
+  peer_->markBlockAsKnown(hash);
+
+  if (dag_blk_mgr_) {
+    if (!dag_blk_mgr_->isBlockKnown(hash) && block_requestes_set_.count(hash) == 0) {
+      block_requestes_set_.insert(hash);
+      requestBlock(packet_data.from_node_id_, hash);
+    }
+  } else if (peers_state_->test_blocks_.find(hash) == peers_state_->test_blocks_.end() &&
+             block_requestes_set_.count(hash) == 0) {
+    block_requestes_set_.insert(hash);
+    requestBlock(packet_data.from_node_id_, hash);
+  }
+}
+
+inline void DagPacketsHandler::processGetNewBlockPacket(const PacketData &packet_data, const dev::RLP &packet_rlp) {
+  blk_hash_t const hash(packet_rlp[0]);
+  peer_->markBlockAsKnown(hash);
+  LOG(log_dg_) << "Received GetNewBlockPacket" << hash.toString();
+
+  if (dag_blk_mgr_) {
+    auto block = db_->getDagBlock(hash);
+    if (block) {
+      sendBlock(packet_data.from_node_id_, *block);
+    } else
+      LOG(log_nf_) << "NO NEW PACKET: " << hash.toString();
```

Comment:
hmm lets have here some proper message, this is very generic and does not say what actually hapenned